lightning/src/ln/peer_handler.rs

```diff
@@ -897,25 +902,31 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 											Ok(x) => x,
 											Err(e) => {
 												match e {
-													msgs::DecodeError::UnknownVersion => return Err(PeerHandleError { no_connection_possible: false }),
-													msgs::DecodeError::UnknownRequiredFeature => {
+													(msgs::DecodeError::UnknownRequiredFeature, _) => {
 														log_gossip!(self.logger, "Got a channel/node announcement with an unknown required feature flag, you may want to update!");
```

Comment:
Not necessarily relevant to the PR, but could this happen for non-gossip messages?

Reply:
Hmm, yes, it could, I think there was a time where it couldn't but now it definitely can at least for TLVs. Lets fix that in a followup.